cpp/oneapi/dal/backend/common.hpp

```diff
@@ -160,11 +160,37 @@ inline void check_if_same_context(const sycl::queue& q1,
 /// Creates `nd_range`, where global size is multiple of local size
 inline sycl::nd_range<1> make_multiple_nd_range_1d(std::int64_t global_size,
                                                    std::int64_t local_size) {
-    const std::int64_t g = dal::detail::integral_cast<std::size_t>(global_size);
-    const std::int64_t l = dal::detail::integral_cast<std::size_t>(local_size);
+    const std::size_t g = dal::detail::integral_cast<std::size_t>(global_size);
+    const std::size_t l = dal::detail::integral_cast<std::size_t>(local_size);
```

Comment:
nip: its common practice to specify the data type only one time in every expression. It helps to avoid problems like the bug you've fixed.

```cpp
const auto var = some_templated_cast<std::actual_type>(another_var)
```

Reply:
fixed